### DISCUSSION TITLE
Working with Frequency COI

### DISCUSSION THREAD

**Discussion by TimThacker** [CURENT/andes#403] [Data Format]
Hantao, I have read through the documentation for ANDES and it appears that COI is built into ANDES, however I am unsure how to access it. 

I am working on a frequency study in the WECC system, and I believe it would be easier to plot and work with the frequency COI. It seems that ss.COI.omega is empty, do I need to manually enter an index for each generator in the case file? Thank you.

**Comment by cuihantao** [comment]
Please take a look at `kundur_coi.xlsx`. In `GENROU`, you specify the `idx` of the linked COI to the `coi` field. 

Then, in the `COI` sheet, you will create the COI instances with the proper `idx` and `name`, among others.

**Reply by TimThacker** [reply]
Hantao, I understand now that COI is implemented as if a new model is added to the system. How can I access the COI data? I am able to plot it by calling 'ss.TDS.plt.plot(ss.COI.omega)' but if I just call 'ss.COI.omega' I am not given something I can convert into a NumPy array. Could you explain the data structure to me?

Thank you!

**Reply by cuihantao** [reply]
These might be useful to you: https://docs.andes.app/en/latest/_examples/ex2.html#variables
